### ISSUE TITLE
UTM froze and not able to reboot Ubuntu 20.04

### ISSUE THREAD

**Issue by MassimilianoAntonello** [utmapp/UTM#4039]
Good afternoon,
I encountered a strange issue: during normal operation with the virtual machine it frozen.
After I close the window and re-open it Ubuntu doesn't boot again and remains on EFI:
<img width="1500" alt="Schermata 2022-05-18 alle 09 59 22" src="https://user-images.githubusercontent.com/41299506/169314078-fc15ed19-c202-4f0c-90fd-c853105b1521.png">
I lost already 2 times (today and yesterday) all my work since I am not able to access anymore to my VM.
I tried to reboot from kernel 5.4.0-110 and 109 without any success.
I tried to run again an old version with 5.4.0-105 today, I worked on it until few hours ago and again it frozen and display the same EFI screen (so I lost again all my work).
Even trying to reboot from kernel 5.4.0-104 doesn't work (no I did not get any update on linux).
I tried also with different versions of UTM: latest (3.2.3), 3.1.15 and also 2.4.1. Same result.
Thanks

**Configuration**
* UTM Version: 3.2.3, 3.1.15 and 2.4.1 
* OS Version: 12.3.1
* Apple Silicon M1 PRO chip

**Comment by MassimilianoAntonello** [comment]
And... it is just happened again... 3 times in a row. Everything completely gone. All the work done dissolved in a bug.
I supported UTM but UTM did not supported me... how sad.
This time again on kernel 5.4.110... freeze and then bye bye VM

**Comment by osy** [comment]
Can you provide either a crash log (if you see one) or get a spindump when it freezes:

open Activity Monitor. Double click on UTM. Click Sample. Copy the output to a .txt file and upload it?

**Comment by MassimilianoAntonello** [comment]
> Can you provide either a crash log (if you see one) or get a spindump when it freezes:
> 
> open Activity Monitor. Double click on UTM. Click Sample. Copy the output to a .txt file and upload it?

Like this?
Thanks...

[Campione di UTM.txt](https://github.com/utmapp/UTM/files/8732446/Campione.di.UTM.txt)

**Comment by osy** [comment]
@MassimilianoAntonello thanks. When you say “freeze” is the UI still responsive? For example can you click on the toolbar icons, change windows, etc? If so please also get a spindump of a process called QEMULauncher and QEMUHelper if you see it. Thanks.

**Comment by MassimilianoAntonello** [comment]
> @MassimilianoAntonello thanks. When you say “freeze” is the UI still responsive? For example can you click on the toolbar icons, change windows, etc? If so please also get a spindump of a process called QEMULauncher and QEMUHelper if you see it. Thanks.

No, unfortunately the only thing I can do is close the VM windows (and so kill the VM)

**Comment by osy** [comment]
Okay so the spindump you provided above is in a state where the window is completely frozen and you can’t click the power button or usb button or cd button etc?

**Comment by MassimilianoAntonello** [comment]
> Okay so the spindump you provided above is in a state where the window is completely frozen and you can’t click the power button or usb button or cd button etc?

Oh sorry! No unfortunately when it frozen I close the windows and when I reopened it it displayed again the issue of the efi and doesn’t proceed further. I did not manage to have the report just when it frozen. Sorry for the misunderstanding :(

**Comment by osy** [comment]
Okay so next time you experience a full freeze (you can tell if there is no response when you click the CD icon), please take a spindump of UTM, QEMULauncher, and QEMUHelper.

**Comment by MassimilianoAntonello** [comment]
> Okay so next time you experience a full freeze (you can tell if there is no response when you click the CD icon), please take a spindump of UTM, QEMULauncher, and QEMUHelper.

Ok so I should try again with a new VM until it will happen again but before closing it I will take the spindump. Let’s hope to solve.
Thanks

**Comment by MassimilianoAntonello** [comment]
> Okay so next time you experience a full freeze (you can tell if there is no response when you click the CD icon), please take a spindump of UTM, QEMULauncher, and QEMUHelper.

What I can add that maybe can be useful is that I cloned the last VM before reinstalling everything and both worked.
After the freeze of one of the two now also the other one has the same boot issue and I can access to that one too.

**Comment by MassimilianoAntonello** [comment]
> Okay so next time you experience a full freeze (you can tell if there is no response when you click the CD icon), please take a spindump of UTM, QEMULauncher, and QEMUHelper.

Hi, new day, new VM, new crash...
But this time before lose everything again I did the spindump.
I hope they could help since I can't anymore working like this...
Thanks for your time and support.

PS the QEMUHelper says something like more than 300GB physical footprint???

[QEMUHelper_Crash_Error.txt](https://github.com/utmapp/UTM/files/8738088/QEMUHelper_Crash_Error.txt)
[QEMULauncher_Crash_Error.txt](https://github.com/utmapp/UTM/files/8738089/QEMULauncher_Crash_Error.txt)
[UTM_Crash_Error.txt](https://github.com/utmapp/UTM/files/8738090/UTM_Crash_Error.txt)

**Comment by osy** [comment]
Are you absolutely certain that the UTM GUI is frozen (I am not talking about inside the VM). When you press the CD icon or USB icon, there is no response? From your spindump files it does not appear that the UI is frozen.

**Comment by MassimilianoAntonello** [comment]
> Are you absolutely certain that the UTM GUI is frozen (I am not talking about inside the VM). When you press the CD icon or USB icon, there is no response? From your spindump files it does not appear that the UI is frozen.

No you are right sorry! This time it was not frozen entirely and after two reboot it worked again! I will do the same dump in case it will happen again that after the freeze I will not be able to restore it.
Thanks again and sorry
